### PR TITLE
Geocode refactor

### PR DESCRIPTION
**Pull request by Jwan622**
-unlimited geocoding now courtesy of bing.

### CODE REVIEW

**Comment by cluhring** [comment]
Should we take down your Seed Refactor Pull Request, as it looks to be merged into this PR with your Bing stuff?

File: spec/factories/techhire_locations.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
```
Comment (cluhring) [inline spec/factories/techhire_locations.rb:1]: needed?

File: spec/models/techhire_location_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe TechhireLocation, type: :model do
```
Comment (cluhring) [inline spec/models/techhire_location_spec.rb:3]: Model tests?  Geocoding?

File: app/helpers/youtube_helper.rb [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+# #equiv to:
+# YoutubeHelper.extend(YoutubeHelper)
+#
+# #equiv to:
+# extend self
+#
+# #equiv to:
+# YoutubeHelper.singleton_class.class_eval do
+#   include YoutubeHelper
+# end
```
Comment (laurawhalin) [inline app/helpers/youtube_helper.rb:25]: Do we need these comments in the app? I'm find with the comment on line 9, but the rest of them just look like personal notes?

File: db/migrate/20150429035807_remove_page_from_pdf.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class RemovePageFromPdf < ActiveRecord::Migration
+  def change
+    remove_column :pdfs, :page
+  end
```
Comment (laurawhalin) [inline db/migrate/20150429035807_remove_page_from_pdf.rb:4]: nice simplification :sun_with_face:

File: spec/factories/techhire_locations.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+  factory :techhire_location do
+    city "MyString"
+state "MyString"
```
Comment (laurawhalin) [inline spec/factories/techhire_locations.rb:4]: indent this line pls

File: spec/models/video_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+  it "should be invalid with a blank url" do
+    video = build(:video, url: "")
+    video.save
+
+    expect(video.errors.full_messages.to_sentence).to eql("Error Please enter a valid url.")
+    expect(video).to be_invalid
+  end
+
+  it "should ve valid with a lot of 1,2,3 or 4" do
```
Comment (laurawhalin) [inline spec/models/video_spec.rb]: small typo

File: spec/models/video_spec.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    video1 = build(:video, slot:1)
+    video2 = build(:video, slot:2)
+    video3 = build(:video, slot:3)
+    video4 = build(:video, slot:4)
```
Comment (laurawhalin) [inline spec/models/video_spec.rb:22]: Change one of these so it doesn't contain a slot to show that it is still valid.

File: spec/models/video_spec.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    video4 = build(:video, page:"learn more")
+    video5 = build(:video, page:"home")
+
+    expect(video1).to be_valid
+    expect(video2).to be_invalid
+    expect(video3).to be_invalid
+    expect(video4).to be_valid
+    expect(video5).to be_valid
+  end
+end
```
Comment (laurawhalin) [inline spec/models/video_spec.rb:52]: :whale:

File: app/helpers/youtube_helper.rb [hunk 9]
```diff
@@ -1,8 +1,25 @@
 module YoutubeHelper
+  # Look below for expanded code
+  class << YoutubeHelper
+    include YoutubeHelper
+    include ActionView::Helpers::TagHelper
+  end
+
```
Comment (scottalexandra) [inline app/helpers/youtube_helper.rb:7]: Nice use of ruby object model! Opening up the included class and making the methods in the module class methods!

File: app/models/techhire_location.rb [hunk 10]
```diff
@@ -0,0 +1,15 @@
+class TechhireLocation < ActiveRecord::Base
+  include ActiveModel::Dirty
+
```
Comment (scottalexandra) [inline app/models/techhire_location.rb:3]: Mr. Fancy Pants!  ActiveModel::Dirty monitoring the state of objects!  :facepunch: !!!! Pow Pow!!!!

File: app/views/static_pages/home.html.erb [hunk 11]
```diff
@@ -6,8 +6,6 @@
   <%= link_to "Tools & Resources", tools_resources_path %>
 </div>
 
-<div class="map"></div>
-
 <div class="embed-container">
-  <%= embed("https://www.youtube.com/watch?v=ZzQ3eBerHfM") %>
+  <%= YoutubeHelper.embed("https://www.youtube.com/watch?v=ZzQ3eBerHfM") %>
 </div>
```
Comment (scottalexandra) [inline app/views/static_pages/home.html.erb:11]: Cool, I see it is clearer what the embed method is doing now that its a class method!

File: spec/models/video_spec.rb [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+  it "should be invalid with a blank url" do
+    video = build(:video, url: "")
+    video.save
+
+    expect(video.errors.full_messages.to_sentence).to eql("Error Please enter a valid url.")
+    expect(video).to be_invalid
+  end
+
+  it "should ve valid with a lot of 1,2,3 or 4" do
+    video1 = build(:video, slot:1)
```
Comment (scottalexandra) [inline spec/models/video_spec.rb:19]: Small typo... did you mean be instead of ve?

File: config/initializers/geocoder.rb [hunk 13]
```diff
@@ -0,0 +1,6 @@
+Geocoder.configure({
+  lookup: :bing,
+  key: ENV['BING_GEOCODE_ID'],
+  api_key: ENV['BING_GEOCODE_ID'],
+  cache: Rails.cache
+})
```
Comment (scottalexandra) [inline config/initializers/geocoder.rb:6]: Yay unlimted requests and caching!!!  :fried_shrimp:

**Pull request merged**